keps/sig-auth/541-external-credential-providers/README.md

```diff
@@ -127,24 +154,37 @@ be readable and executable by the client process.
 `args` specifies extra arguments passed to the executable.
 
 `env` specifies environment variables to pass to the provider. The environment
-variables set in the client process are not passed.
+variables set in the client process are also passed to the provider.
 
 ### Provider input format
 
-```
+```json
 {
-  "apiVersion": "client.authentication.k8s.io/<version>",
-  "kind": "ExecCredential"
+  "apiVersion": "client.authentication.k8s.io/v1beta1",
+  "kind": "ExecCredential",
+  "spec": {
+    "cluster": {
+      "server": "https://1.2.3.4:8080",
+      "certificate-authority": "/etc/kubernetes/ca.pem"
+    }
+  }
 }
 ```
 
-Provider can safely ignore `stdin` since input object doesn't carry any data.
+The `spec.cluster` field is the current cluster that the client is communicating
+with.  This struct is defined in `k8s.io/client-go/tools/clientcmd/api/v1` (it
+is used in the `kubeconfig` file format).  This allows the executable to perform
+different actions based on the current cluster (i.e. get a token for a particular
+cluster).
+
+This data is passed to the executable via the `KUBERNETES_EXEC_INFO` environment
+variable in a JSON serialized object.
```

Comment:
Please review this section carefully as it defines an update to the `ExecCredential` API.

Reply:
Reminder to self: this implies a doc update if accepted.

Reply:
Please link https://pkg.go.dev/k8s.io/client-go/tools/clientcmd/api/v1?tab=doc#Cluster to save reader a search.

Reply:
what if `spec.cluster` is different from the cluster set in kubeconfig context?

Reply:
> Please link https://pkg.go.dev/k8s.io/client-go/tools/clientcmd/api/v1?tab=doc#Cluster to save reader a search.

Done.

> Might be nice to note that the reason for having `$KUBERNETES_EXEC_INFO` is to reserve stdin for user->plugin interactions (e.g., if it's a TTY).

Done.

> what if `spec.cluster` is different from the cluster set in kubeconfig context?

Not sure I understand the question.  `spec.cluster` is taken directly from the `kubeconfig`.

See if the extra explanation I added in the doc helps clarify.